recipes/install.rb

```diff
@@ -150,22 +150,121 @@
   not_if { File.directory?("#{node["kagent"]["dir"]}") }
 end
 
-directory node["kagent"]["etc"]  do
+directory node['data']['dir'] do
+  owner 'root'
+  group 'root'
+  mode '0775'
+  action :create
+  not_if { ::File.directory?(node['data']['dir']) }
+end
+
+directory node['kagent']['data_volume']['root_dir']  do
   owner node["kagent"]["user"]
   group node["kagent"]["group"]
   mode "755"
+  recursive true
   action :create
-  not_if { File.directory?("#{node["kagent"]["etc"]}") }
+  not_if { File.directory?("#{node['kagent']['data_volume']['root_dir']}") }
 end
 
-directory "#{node["kagent"]["etc"]}/state_store" do
-  owner node["kagent"]["user"]
-  group node["kagent"]["group"]
-  mode "770"
+directory node['kagent']['data_volume']['logs']  do
+  owner node['kagent']['user']
+  group node['kagent']['group']
+  mode "755"
+  action :create
+  not_if { File.directory?("#{node['kagent']['data_volume']['logs']}") }
+end
+
+bash 'Move kagent logs to data volume' do
+  user 'root'
+  code <<-EOH
+    set -e
+    mv -f #{node['kagent']['logs']}/* #{node['kagent']['data_volume']['logs']}
+    mv -f #{node['kagent']['logs']} #{node['kagent']['logs']}_deprecated
+  EOH
+  only_if { conda_helpers.is_upgrade }
+  only_if { File.directory?(node['kagent']['logs'])}
+  not_if { File.symlink?(node['kagent']['logs'])}
+end
+
+link node['kagent']['logs'] do
+  owner node['kagent']['user']
+  group node['kagent']['group']
+  mode '0755'
+  to node['kagent']['data_volume']['logs']
+end
+
+directory node['csr']['data_volume']['logs']  do
+  owner node['kagent']['certs_user']
+  group node['kagent']['group']
+  mode "750"
+  action :create
+  not_if { File.directory?(node['csr']['data_volume']['logs']) }
+end
+
+bash 'Move CSR log file to data volume' do
+  user 'root'
+  code <<-EOH
+    set -e
+    mv -f #{node['csr']['log-file']}/* #{node['csr']['data_volume']['log-file']}
```

Comment:
Does this work? You are moving it like it's a directory, but it's actually a file

Reply:
You're right. I'm surprised it didn't fail during my upgrade test.